### PR TITLE
Add proper navigation

### LINKED ISSUE #46: Add navigation links to the navbar
The navigation bar should contain all the links available on the landing page, probably in some sort of dropdown. On mobile, it will likely be necessary to have these in a sidenav

### PR DESCRIPTION
**Pull request by jbritain**
This adds site-wide navigation for both large and small screens. On large screens there are dropdowns in the navbar, on small screens there is a hamburger button which opens a drawer.

On desktop
desktop nav

On mobile (taken on actual hardware since the screenshots did not render properly)
mobile nav closed
mobile nav open

### CODE REVIEW

**Review by nitishdhar — CHANGES_REQUESTED** [review]
Nice, looking good. Can you move the burger menu on the left of the logo when in mobile view? We can discuss on discord if that doesn't make sense. Why does the dropdown menu go flat in the mobile view? I guess it's fine for now but when the navigation content increases, we probably will bring that back.

**Comment by jbritain** [comment]
I think that unless we end up with significantly more pages, it's better to leave the navigation flat, to save taps, and to make it easier to find what you're looking for.

On desktop, you can hover over each item on the navbar to show the relevant list of pages, of course this is not possible on mobile so they would have to tap on each item looking for what they want.

**Comment by jbritain** [comment]
I can move the button to the left, I generally prefer it on the right as it's easier to reach but that's a personal choice, and if Google preferred to put it on the left in all their apps then they were probably onto something.

**Comment by llamaair** [comment]
I agree with jbritain, I also prefer the button on the right as it is easier to reach, and it also looks better when all the buttons are gathered on one side, next to eachother (the light mode button and the burger menu).

**Comment by nitishdhar** [comment]
IMG_1664
IMG_1665

Navigation in non mobile view is on the left, it’s symmetrical to keep it on left in mobile view as well. I agree it looks good with the dark mode icon but that right space is usually reserved for quick actions we might have in future, uploaded some examples for reserving that right real estate.

@jbritain I am good with the flat view of nav items.

**Comment by jbritain** [comment]
Button has been moved!

File: app/_components/nav-drawer/NavDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import {
```
Comment (nitishdhar) [inline app/_components/nav-drawer/NavDrawer.tsx:1]: Can you add some basic component test for this new component, even a unit test that checks rendering is enough. We can add some integration tests to check navigation behavior as a follow up.

**Review by nitishdhar — CHANGES_REQUESTED** [review]
Looks good, just one request to add basic unit tests, nothing complex for now but just something to render test the new components. I will merge once it has the test. Thanks

File: app/_components/nav-drawer/NavDrawer.tsx (hunk 1 above)
Comment (chrismou) [reply app/_components/nav-drawer/NavDrawer.tsx:1]: Added basic render and close/open tests

**Review by chrismou — APPROVED** [review]

**Review by nitishdhar — APPROVED** [review]

**Pull request merged**